core/posts.go

```diff
@@ -459,81 +552,106 @@ func validatePost(post *pb.Post) (err error) {
 			case error:
 				err = x
 			default:
-				err = errors.New("Unknown panic")
+				err = ErrPostUnknownPanic
 			}
 		}
 	}()
 
 	// Slug
 	if post.Slug == "" {
-		return errors.New("Slug must not be empty")
+		return ErrPostSlugNotEmpty
 	}
 	if len(post.Slug) > SentenceMaxCharacters {
-		return fmt.Errorf("Slug is longer than the max of %d", SentenceMaxCharacters)
+		return ErrPostSlugLongerThanMax
 	}
 	if strings.Contains(post.Slug, " ") {
-		return errors.New("Slugs cannot contain spaces")
+		return ErrPostSlugSpaces
 	}
 	if strings.Contains(post.Slug, "/") {
-		return errors.New("Slugs cannot contain file separators")
+		return ErrPostSlugFileSeparators
 	}
 
-	// Tile
-	if post.Title == "" {
-		return errors.New("Post must have a title")
+	// Type
+	if _, ok := pb.Post_PostType_value[post.PostType.String()]; !ok {
+		return ErrPostInvalidType
 	}
-	if len(post.Title) > PostTitleMaxCharacters {
-		return fmt.Errorf("Title is longer than the max of %d", PostTitleMaxCharacters)
+
+	// Status
+	if len(post.Status) > PostStatusMaxCharacters {
+		return ErrPostStatusLongerThanMax
 	}
 
 	// Long Form
 	if len(post.LongForm) > PostLongFormMaxCharacters {
-		return fmt.Errorf("Post is longer than the max of %d characters", PostLongFormMaxCharacters)
+		return ErrPostLongFormLongerThanMax
 	}
 
 	// Tags
 	if len(post.Tags) > MaxPostTags {
-		return fmt.Errorf("Tags in the post is longer than the max of %d characters", MaxPostTags)
+		return ErrPostTagsLongerThanMax
 	}
 	for _, tag := range post.Tags {
 		if tag == "" {
-			return errors.New("Tags must not be empty")
+			return ErrPostTagsEmpty
 		}
 		if len(tag) > PostTagsMaxCharacters {
-			return fmt.Errorf("Tags must be less than max of %d", PostTagsMaxCharacters)
+			return ErrPostTagsLengthLongerThanMax
+		}
+	}
+
+	// Channels
+	if len(post.Channels) > MaxPostChannels {
+		return ErrPostChannelsLongerThanMax
```

Comment:
The error looks like it should be `ErrPostChannelsLengthLongerThanMax`.

Reply:
I see. They're right, just the name allows them to be confused with their similarly named counterpart.